src/fhirConfig.ts

```diff
@@ -3,15 +3,20 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { FhirVersion, TypeOperation, R4Resource, STU3Resource, SystemOperation, ConfigVersion } from './constants';
-import { Persistence } from './persistence';
-import { History } from './history';
-import { Search } from './search';
 import { Authorization } from './authorization';
 import { Bundle } from './bundle';
+import { ConfigVersion, FhirVersion, R4Resource, STU3Resource, SystemOperation, TypeOperation } from './constants';
+import { History } from './history';
+import { Persistence } from './persistence';
+import { Search } from './search';
+
+export interface OAuthStrategy {
+    oauthAuthorizationUrl?: string;
+    oauthTokenUrl?: string;

```

Comment:
Remove the optional flag `?`
```suggestion
    oauthAuthorizationUrl: string;
    oauthTokenUrl: string;
```

Reply:
@rsmayda 
After taking another look maybe it makes more sense to remove all of these and just have openidConfigurationUrl.

Cognito, Okta and Auth0 support it and this url gives you access to all the values that we need.

Okta:  https://${yourOktaDomain}/oauth2/${authServerId}/.well-known/openid-configuration
Cognito:  https://cognito-idp.{region}.amazonaws.com/{userPoolId}/.well-known/openid-configuration
Auth0: https://some-tenant.us.auth0.com/.well-known/openid-configuration

Were you planning on using the jwks uri in order to get the public key for token verification?

Reply:
hmmm great idea one con is it adds a network call to get these urls

> Were you planning on using the jwks uri in order to get the public key for token verification?

Yep or implementers could use token introspection/userinfo urls

Reply:
We should be able to cache no?

Reply:
You are right.  Will make this explicit. 👍